main.go

```diff
@@ -20,16 +20,20 @@ import (
 	"os/exec"
 	"regexp"
 	"runtime"
+	"sync/atomic"
 	"syscall"
 	"time"
 )
 
 var (
-	flagP       = flag.Int("p", runtime.NumCPU(), "run `N` processes in parallel")
-	flagTimeout = flag.Duration("timeout", 10*time.Minute, "timeout each process after `duration`")
-	flagKill    = flag.Bool("kill", true, "kill timed out processes if true, otherwise just print pid (to attach with gdb)")
-	flagFailure = flag.String("failure", "", "fail only if output matches `regexp`")
-	flagIgnore  = flag.String("ignore", "", "ignore failure if output matches `regexp`")
+	flagP             = flag.Int("p", runtime.NumCPU(), "run `N` processes in parallel")
+	flagTimeout       = flag.Duration("timeout", 10*time.Minute, "timeout each process after `duration`")
+	flagKill          = flag.Bool("kill", true, "kill timed out processes if true, otherwise just print pid (to attach with gdb)")
+	flagFailure       = flag.String("failure", "", "fail only if output matches `regexp`")
+	flagIgnore        = flag.String("ignore", "", "ignore failure if output matches `regexp`")
+	flagMaxTime       = flag.Duration("maxtime", 0, "maximum time to run")
+	flagMaxRuns       = flag.Int("maxruns", 0, "maximum number of runs")
+	flagQuitAfterFail = flag.Bool("quitAfterFail", false, "quit after the first detected failure if true")
```

Comment:
`quitonfail`?

On Tue, Feb 2, 2016 at 9:55 AM, Bram Gruneir <notifications@github.com>
wrote:

> In main.go
> <https://github.com/cockroachdb/stress/pull/2#discussion_r51578084>:
>
> >  	"syscall"
> >  	"time"
> >  )
> >
> >  var (
> > -	flagP       = flag.Int("p", runtime.NumCPU(), "run `N` processes in parallel")
> > -	flagTimeout = flag.Duration("timeout", 10*time.Minute, "timeout each process after `duration`")
> > -	flagKill    = flag.Bool("kill", true, "kill timed out processes if true, otherwise just print pid (to attach with gdb)")
> > -	flagFailure = flag.String("failure", "", "fail only if output matches `regexp`")
> > -	flagIgnore  = flag.String("ignore", "", "ignore failure if output matches `regexp`")
> > +	flagP             = flag.Int("p", runtime.NumCPU(), "run `N` processes in parallel")
> > +	flagTimeout       = flag.Duration("timeout", 10*time.Minute, "timeout each process after `duration`")
> > +	flagKill          = flag.Bool("kill", true, "kill timed out processes if true, otherwise just print pid (to attach with gdb)")
> > +	flagFailure       = flag.String("failure", "", "fail only if output matches `regexp`")
> > +	flagIgnore        = flag.String("ignore", "", "ignore failure if output matches `regexp`")
> > +	flagMaxTime       = flag.Duration("maxtime", 0, "maximum time to run")
> > +	flagMaxRuns       = flag.Int("maxruns", 0, "maximum number of runs")
> > +	flagQuitAfterFail = flag.Bool("quitAfterFail", false, "quit after the first detected failure if true")
>
> Oh, it was meant to have no capitalization. I'll fix it.
> And I just couldn't come up with anything better.
> quitafterfirstfailure seemed too long.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/stress/pull/2/files#r51578084>.
>